### DISCUSSION TITLE
Swagger default Enum missing and Optional field also

### DISCUSSION THREAD

**Discussion by raphaelauv** [fastapi/fastapi#7371] [Questions] [answered]
SInce few days the default value provided for an ENUM is no more showing in the swagger

Before :
![Screenshot from 2020-07-23 00-19-47](https://user-images.githubusercontent.com/10202690/88235027-4ee61780-cc7a-11ea-9b9d-b763988f874f.png)

now:
![Screenshot from 2020-07-23 00-20-04](https://user-images.githubusercontent.com/10202690/88235043-560d2580-cc7a-11ea-8544-b678712c2946.png)

```python
@router.post('/a', response_model=Something)
async def endpoint_a(user: Something2,
                       keep_option: KeepOption = KeepOption.KEEP_NO_MISCLASSIFIED_ADS,
                       limit: Optional[int] = None):
```

Also a field limit:Optional[int] = None is not considered optional in the swagger

![Screenshot from 2020-07-23 00-21-23](https://user-images.githubusercontent.com/10202690/88235121-776e1180-cc7a-11ea-8e32-8493409d256e.png)

It's maybe only related to swagger-ui , do you have any idee ? thank you

**Comment by Gatux** [comment, marked as answer]
The problem seems to be in pydantic : https://github.com/samuelcolvin/pydantic/pull/1749

**Comment by PrettyWood** [comment]
Exactly
It seems to be due to [a change of enum schema](https://github.com/samuelcolvin/pydantic/pull/1432), which came in [pydantic v1.6](https://github.com/samuelcolvin/pydantic/releases/tag/v1.6).
@mostaphaRoudsari made a very good statement in https://github.com/samuelcolvin/pydantic/pull/1749#issuecomment-662728965 to fix the doc generation.

**Comment by raphaelauv** [comment]
correct with pydantic 1.7

**Comment by tiangolo** [comment]
Thanks for the discussion everyone! 🍰 

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
